src/SqlPersistence/Outbox/OutboxPersister.cs

```diff
@@ -79,8 +81,14 @@ public async Task<OutboxMessage> Get(string messageId, ContextBag context)
                 command.CommandText = outboxCommands.Get;
                 command.Transaction = transaction;
                 command.AddParameter("MessageId", messageId);
-                // to avoid loading into memory SequentialAccess is required which means each fields needs to be accessed
-                using (var dataReader = await command.ExecuteReaderAsync(CommandBehavior.SingleRow | CommandBehavior.SequentialAccess).ConfigureAwait(false))
+
+                // to avoid loading into memory SequentialAccess is required which means each fields needs to be accessed, but SequentialAccess is unsupported for SQL Server AlwaysEncrypted
+                var behavior = CommandBehavior.SingleRow;
+                if (!connection.IsEncrypted())
```

Comment:
Can we store/cache this value somehow? I am afraid that parsing the connection string each time we execute a query will add significant cost (although not sure how to measure it).

Maybe the dialect could inspect if connection is encrypted first time someone calls `CreateCommand` and then set `IsEncrypted` property on the `CommandWrapper` it returns.